bundles/aliyun-bundle/src/main/java/org/apache/gravitino/oss/credential/OSSTokenProvider.java

```diff
@@ -0,0 +1,267 @@
+    Map<String, Statement.Builder> bucketGetLocationStatementBuilder = new HashMap<>();
+
+    String arnPrefix = getArnPrefix();
+    Stream.concat(readLocations.stream(), writeLocations.stream())
+        .distinct()
+        .forEach(
+            location -> {
+              URI uri = URI.create(location);
+              allowGetObjectStatementBuilder.addResource(getOssUriWithArn(arnPrefix, uri));
+              String bucketArn = arnPrefix + getBucketName(uri);
+              // ListBucket
+              bucketListStatementBuilder.computeIfAbsent(
+                  bucketArn,
+                  key ->
+                      Statement.builder()
+                          .effect(Effect.ALLOW)
+                          .addAction("oss:ListBucket")
+                          .addResource(key)
+                          .condition(getCondition(uri)));
+              // GetBucketLocation
+              bucketGetLocationStatementBuilder.computeIfAbsent(
+                  bucketArn,
+                  key ->
+                      Statement.builder()
+                          .effect(Effect.ALLOW)
+                          .addAction("oss:GetBucketLocation")
+                          .addResource(key));
+            });
+
+    if (!writeLocations.isEmpty()) {
+      Statement.Builder allowPutObjectStatementBuilder =
+          Statement.builder()
+              .effect(Effect.ALLOW)
+              .addAction("oss:PutObject")
+              .addAction("oss:DeleteObject");
+      writeLocations.forEach(
+          location -> {
+            URI uri = URI.create(location);
+            allowPutObjectStatementBuilder.addResource(getOssUriWithArn(arnPrefix, uri));
+          });
+      policyBuilder.addStatement(allowPutObjectStatementBuilder.build());
+    }
+
+    if (!bucketListStatementBuilder.isEmpty()) {
+      bucketListStatementBuilder
+          .values()
+          .forEach(statementBuilder -> policyBuilder.addStatement(statementBuilder.build()));
+    } else {
+      // add list privilege with 0 resources
+      policyBuilder.addStatement(
+          Statement.builder().effect(Effect.ALLOW).addAction("oss:ListBucket").build());
+    }
+    bucketGetLocationStatementBuilder
+        .values()
+        .forEach(statementBuilder -> policyBuilder.addStatement(statementBuilder.build()));
+
+    policyBuilder.addStatement(allowGetObjectStatementBuilder.build());
+    try {
+      return objectMapper.writeValueAsString(policyBuilder.build());
+    } catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Condition getCondition(URI uri) {
+    return Condition.builder()
+        .stringLike(
+            StringLike.builder()
+                .addPrefix(concatPathWithSep(trimLeadingSlash(uri.getPath()), "*", "/"))
+                .build())
+        .build();
+  }
+
+  private String getArnPrefix() {
+    if (StringUtils.isNotEmpty(region)) {
```

Comment:
> Could you run pass the test if setting region explicitly? I failed to run it.

I used 'oss-cn-hangzhou' and it passed the test. Which region are you using?

Reply:
Oh, I missed `oss` in region ID. I could run the pass now. By the way, Should we extract the region information from the endpoint if the user doesn't set the region ID explicitly? And why should the user set the region ID explicitly?

Reply:
Okay, can extract it from the endpoint. I'll make some changes

Reply:
> Okay, can extract it from the endpoint. I'll make some changes

@FANNG1 I think it's better to keep this parameter configuration, mainly because I'm worried that if the rules change, it won't match. What do you think?

Reply:
ok